hudi-common/src/main/java/org/apache/hudi/common/table/view/FileSystemViewManager.java

```diff
@@ -159,12 +160,12 @@ private static HoodieTableFileSystemView createInMemoryFileSystemView(Serializab
     return new HoodieTableFileSystemView(metaClient, timeline, viewConf.isIncrementalTimelineSyncEnabled());
   }
 
-  public static HoodieTableFileSystemView createInMemoryFileSystemView(HoodieTableMetaClient metaClient,
-                                                                       boolean useFileListingFromMetadata,
-                                                                       boolean verifyListings) {
+  public static HoodieTableFileSystemView createInMemoryFileSystemViewForReaders(HoodieTableMetaClient metaClient,
+      boolean useFileListingFromMetadata, boolean verifyListings) {
     LOG.info("Creating InMemory based view for basePath " + metaClient.getBasePath());
     if (useFileListingFromMetadata) {
-      return new HoodieMetadataFileSystemView(metaClient,
+      HoodieMREngineContext engineContext = new HoodieMREngineContext(metaClient.getHadoopConf());
```

Comment:
@umehrot2 hello, why not use spark EngineContext? Because hudi can run just use spark, many users environment just have spark , but not have mr.

Reply:
Hudi has added support for other engines like Flink. Also HoodieSparkEngineContext cannot be used here this API is called inside individual map reduce tasks from engines like spark and hive. In case of spark, the individual tasks do not have access to spark context. The spark context is not serializable.

Reply:
On a side note, we should move it to the hudi-hadoop-mr module or rename this to `HoodieLocalEngine` or something.  Lets discuss the relocation in the other comment.